week2/task7/Program.cs

```diff
@@ -0,0 +1,103 @@
+﻿var signaler = new Signaler();
+signaler.AddTime(new JupiterTime(2, 00));
+signaler.AddTime(new JupiterTime(4, 00));
+signaler.AddTime(new JupiterTime(6, 00));
+signaler.Check(new JupiterTime(4, 21));
+
+class JupiterTime
+{
+    public int _hours = 0, _minutes = 0;
+
+    public override string ToString()
+    {
+        if (_minutes < 10) return _hours + ":0" + _minutes;
+        return _hours + ":" + _minutes;
+    }
+    public int Hours
+    {
+        get
+        {
+            return _hours;
+        }
+        set
+        {
+            if (value < 0) _hours = (10 + (value % 10));
+            else if (value > 9) _hours = (value % 10);
+            else _hours = value;
+        }
+    }
+    public int Minutes
+    {
+        get
+        {
+            return _minutes;
+        }
+        set
+        {
+            if (value < 0)
+            {
+                _hours = _hours + (value / 60);
+                if (_hours < 0) _hours += 10;
+                _minutes = _minutes + (value % 60);
+                if (_minutes < 0)
+                {
+                    _minutes += 60;
+                    _hours -= 1;
+                }
+            }
+            else if (value > 59)
+            {
+                _hours = _hours + (value / 60);
+                if (_hours > 9) _hours -= 10;
+                _minutes = (value % 60);
+            }
+            else _minutes = value;
+        }
+    }
+
+    public JupiterTime(int hours, int minutes)
+    {
+        Hours = hours;
+        Minutes = minutes;
+    }
+    public JupiterTime AddHours(int number)
+    {
+        Hours = _hours + number;
+        return new JupiterTime(_hours, _minutes);
+    }
+    public JupiterTime AddMinutes(int number)
+    {
+        Minutes = number;
+        return new JupiterTime(_hours, _minutes);
+    }
+}
+
+class Signaler
+{
+    List<JupiterTime> Timelist = new List<JupiterTime>();
+
+    public void AddTime(JupiterTime time)
+    {
+        Timelist.Add(time);
+    }
+
+    public void Check(JupiterTime time)
```

Comment:
The logic behind this method should not be tied to the length of the string. Strongly typed objects are easier to use when validating logic. 
Besides this, when changing the amount of signals in the array, this logic would not work anymore and it would have to be updated. 
To avoid this, a more suitable approach would be to use foreach loops and if statements to check the hours and minutes. 

e.g. for logic

// loop over the signals
   // is the signal hour after the time? => add the time to a new list
   // if it is the same hour, are the minutes passed?  => add the time to a new list
//

// are there any items in the new list?
   // loop over the times and console log
// if there are no times, then print the message from the task

Reply:
Let me know if I should explain further or if something doesnt make sense. 
p.s.: debugger is your best friend in c# 😄 (F5)